Problem with parsing multi dashes

Hi,

I have configMaps which keep data with multiple dashes. It raises an error when I try to use kube-score with that.

```>> cat test_configmap.yaml
---

apiVersion: v1
kind: ConfigMap
metadata:
  name: awesome
data:
  awesome.crt: |
    -----BEGIN CERTIFICATE-----
    TEST_DATA
    -----END CERTIFICATE-----
---
>> kube-score test_configmap.yaml
panic: yaml: unmarshal errors:
  line 1: cannot unmarshal !!str `TEST_DA...` into score.detectKind

goroutine 1 [running]:
main.main()
        /path/cmd/kube-score/main.go:78 +0x17c5
```

--- a/score/score.go
+++ b/score/score.go
@@ -101,7 +101,7 @@ func Score(config Configuration) (*scorecard.Scorecard, error) {
 		// Convert to unix style newlines
 		fullFile = bytes.Replace(fullFile, []byte("\r\n"), []byte("\n"), -1)
 
-		for _, fileContents := range bytes.Split(fullFile, []byte("---\n")) {
+		for _, fileContents := range bytes.Split(fullFile, []byte("\n---\n")) {
 			var detect detectKind
 			err = yaml.Unmarshal(fileContents, &detect)
 			if err != nil {

--- a/score/score_test.go
+++ b/score/score_test.go
@@ -20,8 +20,7 @@ func testFile(name string) *os.File {
 // testExpectedScoreWithConfig runs all tests, but makes sure that the test for "testcase" was executed, and that
 // the grade is set to expectedScore. The function returns the comments of "testcase".
 func testExpectedScoreWithConfig(t *testing.T, config Configuration, filename string, testcase string, expectedScore scorecard.Grade) []scorecard.TestScoreComment {
-	config.AllFiles = []io.Reader{testFile(filename)}
-	sc, err := Score(config)
+	sc, err := testScore(config, filename)
 	assert.NoError(t, err)
 
 	for _, objectScore := range sc.Scores {
@@ -35,6 +34,11 @@ func testExpectedScoreWithConfig(t *testing.T, config Configuration, filename st
 
 	t.Error("Was not tested")
 	return nil
+}
+
+func testScore(config Configuration, filename string) (*scorecard.Scorecard, error) {
+	config.AllFiles = []io.Reader{testFile(filename)}
+	return Score(config)
 }
 
 func testExpectedScore(t *testing.T, filename string, testcase string, expectedScore scorecard.Grade) []scorecard.TestScoreComment {
@@ -115,4 +119,9 @@ func TestContainerSecurityContextLowGroup(t *testing.T) {
 
 func TestContainerSecurityContextHighIds(t *testing.T) {
 	testExpectedScore(t, "pod-security-context-high-ids.yaml", "Container Security Context", 10)
+}
+
+func TestConfigMapMultiDash(t *testing.T) {
+	_, err := testScore(Configuration{}, "configmap-multi-dash.yaml")
+	assert.Nil(t, err)
 }

--- a/score/testdata/configmap-multi-dash.yaml
+++ b/score/testdata/configmap-multi-dash.yaml
@@ -1,0 +1,11 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: awesome
+data:
+  awesome.crt: |
+    -----BEGIN CERTIFICATE-----
+    TEST_DATA
+    -----END CERTIFICATE-----
+---

